src/Compilers/CSharp/Portable/Syntax/NullableContextStateMap.cs

```diff
@@ -19,12 +19,16 @@ namespace Microsoft.CodeAnalysis.CSharp.Syntax
         internal int Position { get; }
         internal bool? WarningsState { get; }
         internal bool? AnnotationsState { get; }
+        internal bool AnnotationsExplicitlyRestored { get; }
```

Comment:
>AnnotationsExplicitlyRestored [](start = 22, length = 29)

I think it would be more efficient and easier for consumers if we instead moved to four state byte-based enum (like ThreeState) with values: Unknown, Enabled, Disabled, Restored.